### DISCUSSION TITLE
Inquiry on Modifying Anura3D to Import FEM Stress, Saturation, and Pore Pressure

### DISCUSSION THREAD

**Discussion by stig-cz** [Anura3D/Anura3D_OpenSource#260] [Q&A]
Hi everyone,

I'm currently working with Anura3D for slope failure simulations and am exploring the possibility of modifying the code to import variables (such as stress, strain, saturation, and pore water pressure) from external FEM simulations like PLAXIS into the MPM stage of Anura3D. The idea is to initialize the material points in Anura3D with these pre-computed FEM results.

I’d like to know if it’s feasible to implement this in Anura3D and, if so, which files or modules would be most relevant to modify. Additionally, any advice on the best approach or potential challenges to keep in mind would be greatly appreciated.

Thank you in advance for any insights or guidance!

Best regards,
Chongzhi Wu

**Comment by lurpi** [comment]
Yes, you can, this is a quick comment, to get you started, not a full answer.

If you look in the [2024 tutorial]( https://github.com/Anura3D/Anura3D_OpenSource/tree/main/manuals ), you will find
13.4 _Stress initialization from file_
and 
13.3.1.7 after the figures something it is explicitly mentioned _Phreatic Surface → Water table from file PSF_1_

**Reply by elbarrendero** [reply]
A colleague of my university wanted the same. Finally ended up calibrating the model at Stage 0 to obtain similar stress and strain results, and modelled the surface of failure pre-determined by the FEM analysis. This way he obtained after failure displacement. Using K0 procedure. Although lately i found out K0 procedure works best only on homogeneous models (1 material).
